plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeCdfPageSink.java

```diff
@@ -0,0 +1,127 @@
+import io.trino.hdfs.HdfsEnvironment;
+import io.trino.spi.PageIndexerFactory;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.TypeManager;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.OptionalInt;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.airlift.slice.Slices.wrappedBuffer;
+import static io.trino.plugin.deltalake.DeltaLakeColumnType.REGULAR;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+
+public class DeltaLakeCdfPageSink
+        extends AbstractDeltaLakePageSink
+{
+    public static final String CHANGE_TYPE_COLUMN_NAME = "_change_type";
+    public static final String CHANGE_DATA_FOLDER_NAME = "_change_data";
+
+    private final String tableLocation;
+
+    private final ImmutableList.Builder<DataFileInfo> dataFileInfos = ImmutableList.builder();
+
+    public DeltaLakeCdfPageSink(
+            List<DeltaLakeColumnHandle> inputColumns,
+            List<String> originalPartitionColumns,
+            PageIndexerFactory pageIndexerFactory,
+            HdfsEnvironment hdfsEnvironment,
+            TrinoFileSystemFactory fileSystemFactory,
+            int maxOpenWriters,
+            JsonCodec<DataFileInfo> dataFileInfoCodec,
+            String outputPath,
+            String tableLocation,
+            ConnectorSession session,
+            DeltaLakeWriterStats stats,
+            TypeManager typeManager,
+            String trinoVersion)
+    {
+        super(
+                inputColumns,
+                originalPartitionColumns,
+                pageIndexerFactory,
+                hdfsEnvironment,
+                fileSystemFactory,
+                maxOpenWriters,
+                dataFileInfoCodec,
+                outputPath,
+                session,
+                stats,
+                typeManager,
+                trinoVersion);
+
+        this.tableLocation = tableLocation;
+    }
+
+    @Override
+    protected void processSynthesizedColumn(DeltaLakeColumnHandle column) {}
+
+    @Override
+    protected void addSpecialColumns(
+            List<DeltaLakeColumnHandle> inputColumns,
+            ImmutableList.Builder<DeltaLakeColumnHandle> dataColumnHandles,
+            ImmutableList.Builder<Integer> dataColumnsInputIndices,
+            ImmutableList.Builder<String> dataColumnNames,
+            ImmutableList.Builder<Type> dataColumnTypes)
+    {
+        dataColumnHandles.add(new DeltaLakeColumnHandle(
+                CHANGE_TYPE_COLUMN_NAME,
+                VARCHAR,
+                OptionalInt.empty(),
+                CHANGE_TYPE_COLUMN_NAME,
+                VARCHAR,
+                REGULAR));
+        dataColumnsInputIndices.add(inputColumns.size());
+        dataColumnNames.add(CHANGE_TYPE_COLUMN_NAME);
+        dataColumnTypes.add(VARCHAR);
+    }
+
+    @Override
+    protected String getRootTableLocation()
+    {
+        return tableLocation;
```

Comment:
This feels like it should be a parameter to the `super` constructor, rather than a method